### DISCUSSION TITLE
[Suggestion] An confirmation for /trust | Build only.

### DISCUSSION THREAD

**Discussion by Remski01** [GriefPrevention/GriefPrevention#252] [Ideas]
[Suggestion] 1:
Could you add an confirmation thing. That the player as example get an message that i can set and that the player need to repeat the command. And maybe that this can be disabled in the config.

[Suggestion] 2:
Could you add an permission where players can only place blocks and remove and can not open chesten in the claim(s).

**Comment by Remski01** [comment]
And uh. Can you add an option to disable that everyone have permission for creating an claim. And add an permission for createing claims? (Im not talking about /claim but with the shovel)

**Comment by RoboMWM** [comment]
Permission for creating claims also applies to shovel usage.

**Comment by Remski01** [comment]
Wel.. So far o noticed it did not worked as it should. But i think its because GP really hate negative permissions. As example. When i add the permission griefprevention.createclaims as an negative permission in world B and not in world A. The player will be enable to createclaims also in world B. And when i ise the negaive permission in world A he can not create an claim in any world. As side note. I use Pex.

World A: Head world.
World B: Non-Head World.

**Comment by RoboMWM** [comment]
PEX is no longer maintained, but it's rather that negative permissions work quite oddly. There is no "plugin" that "hates" negative permissions - only the permission plugin deals with permission granting or not. So this is likely an issue with your configuration.

I've had issues with negative permissions (with PEX at least) in the past, and so I instead use a default "group" in which I put negative permissions in a separate group that inherit from this "default." See here for more details regarding my system: https://github.com/TechFortress/GriefPrevention/issues/28#issuecomment-255556718

**Comment by Remski01** [comment]
Yes. Pex is not longer maintained. But it still works so far.

And what you said: "See here for more details regarding my system:" That work if i want it globaly. But the issue is that i do not want it globaly. That why i used the world A and B. I am trying to get an world based instead globaly. But Im going play with it at our test server when i can. And say what the result's are.

**Comment by RoboMWM** [comment]
> That work if i want it globaly. But the issue is that i do not want it globaly.

Groups can be applied per-world, as well as the individual permissions...

**Comment by Remski01** [comment]
Forget the "And uh. Can you add an option to disable that everyone have permission for creating an claim. And add an permission for createing claims? (Im not talking about /claim but with the shovel)" We soon use Claims as default protection tool.

**Comment by bigpresh** [comment]
I don't much see the point in the confirmation - it'd be needlessly annoying.  What commands would it apply to?  If it applied to `/trust`, say, what would happen if e.g. you were standing in one claim, did the command and it said to do it again to verify, walked to a neighbouring claim, then repeated the command - which claim should it apply to, the one you first tried to do it in, or the one you're in when you do it again to confirm?

It all strikes me as something of a bad idea.  If you're trying to customise messages, you can already do that; if you're trying to make sure people understand what they've done, maybe customise the response to /trust to include extra text, e.g. "(you've allowed them to destroy/grief your build and steal your stuff; if you didn't want that, then /untrust them!)"?

As for the "able to place/destroy blocks but not open chests" permission - why?  Surely if they can break blocks, they can just break the damn chest and take all the items that drop, therefore it would be pointless.


It's not my decision to veto, but I would personally argue that the confirmation for `/trust` it adds more code complexity and odd edge cases and more user friction than it solves, so would vote against it, and don't see the use of a "can place/destroy blocks but not open chests" permission either.
